File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBleBeepModel.swift
//
//  IDOBleBeepModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/19.
//

import Foundation

// MARK: - IDOBleBeepModel
@objcMembers
public class IDOBleBeepModel: NSObject, IDOBaseModel {
    private let version: Int
    /// Error code, 0 for success, non-zero for failure
    public var errCode: Int
    private let itemCount: Int
    public var items: [IDOBleBeepItem]?
    
    enum CodingKeys: String, CodingKey {
        case version = "version"
        case errCode = "err_code"
        case itemCount = "item_count"
        case items = "item"
    }
    
    public init(errCode: Int, items: [IDOBleBeepItem]) {
        self.version = 0
        self.errCode = errCode
        self.itemCount = items.count
        self.items = items
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOItem
@objcMembers
public class IDOBleBeepItem: NSObject, Codable {
    public var name: String
    
    enum CodingKeys: String, CodingKey {
        case name = "name"
    }
    
    public init(name: String) {
        self.name = name
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBleBeepModel.kt
//
//  IDOBleBeepModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///

// MARK: - IDOBleBeepModel

open class IDOBleBeepModel(errCode: Int, items: List<IDOBleBeepItem>) :
    IDOBaseAdapterModel<IDOBleBeepModel> {
    @SerializedName("version")
    private var version: Int = 0
    @SerializedName("item_count")
    private var itemCount: Int = items.size

    /// Error code, 0 for success, non-zero for failure
    @SerializedName("err_code")
    var errCode: Int = errCode
    @SerializedName("item")
    var items: List<IDOBleBeepItem> = items


    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}

// MARK: - IDOItem
open class IDOBleBeepItem(name: String) {
    @SerializedName("name")
    var name: String = name
}
